app/models/server/models/Users.js

```diff
@@ -53,6 +53,7 @@ export class Users extends Base {
 		this.tryEnsureIndex({ 'services.saml.inResponseTo': 1 });
 		this.tryEnsureIndex({ openBusinessHours: 1 }, { sparse: true });
 		this.tryEnsureIndex({ statusLivechat: 1 }, { sparse: true });
+		this.tryEnsureIndex({ extension: 1 }, { sparse: true });
```

Comment:
Can we also add definitions for this new property in the user model i.e. [here](https://github.com/RocketChat/Rocket.Chat/blob/c54ef380d29f99bfde38c31f8dccbf841ffef455/definition/IUser.ts#L110)

Reply:
@Amol, will these extensions always be unique for each user doc? If yes, then I'd recommended adding a `unique` index here as well for further optimizations during search

Reply:
Yes. These extensions will be unique for each agent/user. i.e 2 user's do not share the same extension. So I shall add unique index. Thanks for that suggestion.

Added the property in new user model IUser.